wsr-tic-tac-toe/src/redux/reducers/index.ts

```diff
@@ -0,0 +1,137 @@
+            for (let i: number = 0; i < boardSize; i++) {
+                let tempSet = new Set<number>();
+                for (let j: number = 0; j < boardSize; j++) {
+                    tempSet.add(boardSize * i + j);
+                }
+                WSh = [...WSh, tempSet];
+            }
+
+            for (let i: number = 0; i < boardSize; i++) {
+                let tempSet = new Set<number>();
+                for (let j: number = 0; j < boardSize; j++) {
+                    tempSet.add(boardSize * j + i);
+                }
+                WSv = [...WSv, tempSet];
+            }
+
+            let diagonalSet1 = new Set();
+            let diagonalSet2 = new Set();
+            for (let i = 0; i < boardSize; i++) {
+                diagonalSet1.add(i * boardSize + i);
+                diagonalSet2.add((i + 1) * (boardSize - 1));
+            }
+            let winningScenarios = [];
+            winningScenarios = [
+                ...WSh.map((set) => Array.from(set)),
+                ...WSv.map((set) => Array.from(set)),
+                Array.from(diagonalSet1),
+                Array.from(diagonalSet2),
+            ];
+            return [...winningScenarios];
+        }
+        default:
+            return state;
+    }
+}
+
+const initialBoardState: BoardState = {
+    boardSize: 0,
+    created: false,
+    squares: [],
+    turnOrder: 'player1',
+    gameState: GAME_STATE.ONGOING
+};
+
+export function boardReducer(state = initialBoardState, action: BoardStateActionTypes) {
+    switch (action.type) {
+        case CREATE_BOARD: {
+            let boardSize = action.payload;
+            let initializedSquares: SquareState[] = [];
+            // let squares: SquareState[][] = [];
+            for (let i: number = 0; i < boardSize * boardSize; i++) {
+                initializedSquares = [
+                    ...initializedSquares,
+                    { selected: false, id: i, selectedBy: '' },
+                ]; // creating a 1D array of squares with {selected: false, id: 0,...,boardSize}
+            }
+            return {
+                ...state,
+                boardSize: boardSize,
+                squares: initializedSquares,
+                created: true,
+            };
+        }
+        case REMOVE_BOARD: {
+            return initialBoardState; // return initialBoardState
+        }
+        case CLEAR_BOARD: { // kinda unclear how I should do this with {...state,...} and not Object.assign (though the end result should be the same)
+            return Object.assign(
+                {},
+                state,
+                { turnOrder: 'player1' },
+                { gameState: GAME_STATE.ONGOING},
+                {
+                    squares: state.squares.map((square: SquareState) => {
+                        if (square.selected)
+                            return { ...square, selected: false, selectedBy: false };
```

Comment:
selectedBy - neither empty string nor false is correct type to indicate empty value.
Use null in such cases

Reply:
also, you probably don't need both "selected" and "selectedBy", since they are very connected.
Just selectedBy would tell if the square is selected or not (if it's filled - selected, if it's empty - not)